### ISSUE TITLE
Don't catch legitimate uses of (require 'cl)

### ISSUE THREAD

**Issue by Fanael** [purcell/package-lint#4]
It'd be nice if (when it's implemented) the `cl-lib` vs `cl` check didn't warn on legitimate uses of `(require 'cl)`.

I just found one, likely and hopefully the only one, reason to ever `(require 'cl)`: generalized variables. Emacs 24.3 has them in `gv`, which is autoloaded, Emacs <= 24.2 has them in `cl`. Therefore

``` emacs-lisp
(eval-when-compile
  (unless (require 'gv nil t)
    (require 'cl)))
```

makes generalized variables usable everywhere.

**Comment by purcell** [comment]
Closing because at this point I don't think it's worthwhile to add special support for this in those old Emacs versions.
